### PR TITLE
ZEPPELIN-825: Set appropriate default permissions and allow user to c…

### PR DESCRIPTION
**Pull request by rohannr**
### What is this PR for?
This PR makes users who set only r/w permissions the effective owners of the note.
1. If you set readers, and writers and owners is empty, it is automatically set to the user requesting the change.
2. If you set writers, and owners is empty, it is set it to the user requesting the change.
It also fixes a bug that did not allow one to clear the permissions for a note. 

### What type of PR is it?
Bug Fix, Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-825

### How should this be tested?
Outline the steps to test the PR here.

### Questions:
* Does this needs documentation? YES

…lear permissions

This change makes users who set permissions the effective owners of the note.
1. If you set readers, and writers and owners is empty, it is automatically set to the user requesting the change.
2. If you set writers, and owners is empty, it is set it to the user requesting the change.

### CODE REVIEW

File: zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.Status;
 
+import com.google.common.collect.Sets;
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java]: please sort this according to the existing order

**Comment by felixcheung** [comment]
There is another PR on this same bug?
https://github.com/apache/incubator-zeppelin/pull/846

**Comment by rohannr** [comment]
correct, I was in the process of submitting a fix to the JIRA that I filed when that got submitted. This PR also addresses the issue of default owner/writer permissions when they are not set and includes unit tests for both issues.

**Comment by felixcheung** [comment]
Should this be a different JIRA then?

**Comment by Leemoonsoo** [comment]
Looks good. Thanks @rohannr for good improvement. 
Could you create new JIRA issue as @felixcheung suggested and rebase to resolve conflict?

**Comment by rohannr** [comment]
Created a new JIRA (ZEPPELIN-825) and resolved conflict. Change is ready to merge

**Comment by Leemoonsoo** [comment]
Thanks @rohannr for the improvement and test. LGTM

**Comment by Leemoonsoo** [comment]
Merge if there're no more discussions

**Pull request closed without merging**